### PR TITLE
Add version to discovery node toXContent and toString

### PR DESCRIPTION
**Pull request by idegtiarenko**

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
LGTM (as long as CI is happy)

File: server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java [hunk 1]
```diff
@@ -189,39 +185,58 @@ public void testDiscoveryNodeToXContent() {
...
 
-        final String jsonString = Strings.toString(node, randomBoolean(), randomBoolean());
-        final Map<String, Object> topLevelMap = XContentHelper.convertToMap(XContentType.JSON.xContent(), jsonString, randomBoolean());
-
-        assertThat(topLevelMap.toString(), topLevelMap.size(), equalTo(1));
-        assertTrue(topLevelMap.toString(), topLevelMap.containsKey("test-id"));
-
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> detailsMap = (Map<String, Object>) topLevelMap.get("test-id");
+        assertThat(Strings.toString(node, true, false), equalTo(String.format(Locale.ROOT, """
```
Comment (arteam) [inline server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java:194]: Nit: We can use `Strings.format` here without specifying the locale.

Comment (idegtiarenko) [reply server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java]: I was assuming this it not allowed by the check-style.

https://github.com/elastic/elasticsearch/blob/a00a85cd8f5a5e98d89357a6700e7c3de78f690d/build-tools-internal/src/main/resources/forbidden/es-server-signatures.txt#L140-L142

Seems this is okay to do in the test code

Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java]: `String#format` is forbidden; `Strings#format` is good.

Comment (idegtiarenko) [reply server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java]: Okay, I just confirmed this is enforced: https://gradle-enterprise.elastic.co/s/unbjcotx5tcs4

Comment (idegtiarenko) [reply server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java]: Missed that, thanks all!

Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java]: I opened #93336

**Pull request merged**